cupy/cuda/memory.pyx

```diff
@@ -43,6 +47,57 @@ cdef class Memory:
         return self.ptr
 
 
+cdef class ManagedMemory(Memory):
+
+    """Managed memory allocation on a CUDA device.
+
+    This class provides an RAII interface of the CUDA managed memory
+    allocation.
+
+    Args:
+        device (cupy.cuda.Device): Device whose memory the pointer refers to.
+        size (int): Size of the memory allocation in bytes.
+
+    """
+
+    def __init__(self, Py_ssize_t size):
+        self.size = size
+        self.device = None
+        self.ptr = 0
+        if size > 0:
+            self.device = device_mod.Device()
+            self.ptr = runtime.mallocManaged(size)
+            self._support_prefetch_advise = (
+                _cuda_version >= 8000 and
+                int(self.device.compute_capability) >= 60)
+
+    cpdef prefetch(self, stream):
+        """ Prefetch.
+
+        Args:
+            stream (cupy.cuda.Stream): Stream
+
+        """
+        if not self._support_prefetch_advise:
```

Comment:
How about adding argument to control behavior on unsupported situation? Such as `error='ignoer'`.